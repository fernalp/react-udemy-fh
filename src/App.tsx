import { useState } from 'react'
// import reactLogo from './assets/react.svg'
// import viteLogo from '/vite.svg'
import './App.css'
import Footer from './Components/Footer/Footer'
import { Header } from './Components/Header/Header'
import Counter from './Components/Counter/Counter'
import List from './Components/List/List'

function App() {
  const [count, setCount] = useState(0);

  return (
    <>
      {/* <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p> */}
      <Header title='Title desde props'> Yo soy el Hijo </Header>
      <Counter setCount={setCount}>Counter is {count}</Counter>
      <List items={["Cofee", "Steak", "Code"]} render={
        (item: string) => (<span className='gold'>{item}</span>)
      } />
      <Footer business='IngBio' />
    </>
  )
}

export default App
