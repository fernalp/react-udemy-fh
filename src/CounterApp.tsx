// import { number } from "prop-types";
import { useState } from "react"

type Prop = {
    value: number
}

export const CounterApp = ({value}:Prop) => {

    const [counter, setCounter] = useState(value);
    const changeCounter = (n:number):void=>{
        setCounter(counter + n)
    }
    const resetCounter = ()=>{
        setCounter(value)
    }


    return (
        <>
            <h1>Counter:</h1>
            <h4>{counter}</h4>
            <button onClick={()=> changeCounter(1)}>+1</button>
            <button onClick={()=>changeCounter(-1)}>-1</button>
            <button onClick={resetCounter}>Reset</button>
        </>
    )
}
