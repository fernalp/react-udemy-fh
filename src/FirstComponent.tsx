import PropTypes from "prop-types";



type PropsFirstComponent = {
  title: string,
  subTitle: number,
  name: string
}

export const FirstComponent = ({ title, subTitle, name }: PropsFirstComponent) => {
  return (
    <>
      <h1 data-testid="test-title" >{title}</h1>
      <p>{subTitle / 2}</p>
      <p>{name}</p>
      <div>FirstComponent</div>
    </>
  )
}

FirstComponent.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number
}


FirstComponent.defaultProps = {
  title: "No Title",
  subtitle: 21,
  name: "No Name"
}
