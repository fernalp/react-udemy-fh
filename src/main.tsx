import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App.tsx'
import './index.css'
// import { FirstComponent } from './FirstComponent.tsx'
// import { CounterApp } from './CounterApp.tsx'
import { FirstComponent } from './FirstComponent';

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    {/* <CounterApp value={0} />  */}
    <App />
    <FirstComponent subTitle={7} name='Fernando Almanza' title='Soy el mejor' />
    <FirstComponent subTitle={7} title='Soy el mejor' />
  </React.StrictMode>,
)
