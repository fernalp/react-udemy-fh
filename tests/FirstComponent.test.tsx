import { describe, expect, test } from "@jest/globals";
import { render } from "@testing-library/react";
import { FirstComponent } from "../src/FirstComponent";


describe("Pruebas en componente First Component", () => {
    test("Debe Hacer Match con el Snapshot", () => {
        const { container } = render(<FirstComponent subTitle={4} />);
        console.log(container);

        expect(container).toMatchSnapshot();

    });

    test("Debe mostrar el titulo en un h1", () => {
        const title = "Hello World";
        const { getByText } = render(<FirstComponent subTitle={4} title={title} />);

        expect(getByText(title)).toBeTruthy();
    })
})