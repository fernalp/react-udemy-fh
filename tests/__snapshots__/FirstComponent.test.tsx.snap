// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente First Component Debe Hacer Match con el Snapshot 1`] = `
<div>
  <h1>
    No Title
  </h1>
  <p>
    2
  </p>
  <p>
    No Name
  </p>
  <div>
    FirstComponent
  </div>
</div>
`;
